code/script.js

```diff
@@ -244,100 +296,155 @@ const generateBoard = () => {
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
-    }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
-    } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+// When the user made their guess,
+// filter the choices,
+// and redraw the board based on who remains
+const filterCharacters = (guess) => {
+  const { attribute, value, category } = currentQuestion
+    if (category === 'accessories') {
+      if (guess) {
+        alert(`✔ Yes, the person wears ${attribute}! Good guess.
+🤖 Keeping those who wear ${attribute}.`)
+      } else {
+        alert(`❌ Nope, the person doesn't wear ${attribute}! Guess again.
+🤖 Removing those who wear ${attribute}.`)
+        }
+    } else if (category === 'other') {
+      if (guess) {
+        alert(`✔ Yes, ${attribute} is the name of the game! Good guess.
+🤖 Keeping those ones.`)
+      } else {
+        alert(`❌ Nope, ${attribute} isn't in the cards! Guess again.
+🤖 Removing those ones.`)
+        }
+    } else if (category === 'hair color') {
+      if (guess) {
+        alert(`✔ Yes, the person has ${value} hair! Good guess.
+🤖 Keeping every ${value} haired ones.`)
+      } else {
+        alert(`❌ Nope, the person doesn't have ${value} hair! Guess again.
+🤖 Removing those with ${value} hair.`)
+        }
+    } else if (category === 'eye color') {
+      if (guess) {
+        alert(`✔ Yes, gaze into their ${value} eyes! Good guess.
+🤖 Keeping the ${value} eyed ones.`)
+      } else {
+        alert(`❌ Nope, the person doesn't have ${value} eyes!
+🤖 Removing everyone with ${value} eyes.`)
+        }
     }
+  if (guess) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
   }
-
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+  else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// Confirm with the user if they meant to guess
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  confirm(`🙈 You're about to guess on ${suspect}. Are you sure?`)
+  if (confirm) {
+    checkMyGuess(suspect)
```

Comment:
If you don't confirm, it will still make the guess 😬 Try saving the confirm in a variable and use that, something like:
```
const makeAGuess = confirm(`🙈 You're about to guess on ${suspect}. Are you sure?`)
  if (makeAGuess) {
    checkMyGuess(suspect)
  }
```

Reply:
Excellent, I didn't QA this properly I see... This has been fixed & pushed. Thank you! :D